scripts/build_dependencies/build_utils.js

```diff
@@ -0,0 +1,85 @@
+        try {
+            var list_output = child_process.execSync("npm list -g --depth 0 " + package_name, {encoding: "utf8"});
+            if (list_output.indexOf(package_name + "@" + package_version) !== -1) {
+                console.log("Found " + package_name + " in node list!");
+                return true;
+            }
+        } catch (error) {}
+
+        console.log("Could not find " + package_name + " version " + package_version);
+        return false;
+    },
+
+    /**
+     * Downloads an NPM package for use dynamically - so it will only be loaded and used for this single script.
+     * What it does under the hood is check for the npm package in the node modules, then in the npm list, and if
+     * it does not find it in either location, it will download the package at that point, adding it as a dev-dependency.
+     *
+     * It it downloads it dynamically, then it will return true. This is so you can delete the package once you are done,
+     * so it doesn't leave a zombie package in your node_modules. To remove the package, call npmDeleteRuntimePackage
+     *
+     * @param {*} package_name The name of the package you want to download (example: 'cmake-js')
+     * @param {*} package_version The version of the package to download - leave blank for latest. (example: '6.3.2')
+     * @returns True if the package was downloaded dynamically, otherwise false.
+     */
+    npmDownloadAndInstallRuntimePackage : function(package_name, package_version) {
+        console.log("Looking for " + package_name + " version " + package_version + " as a dependency...");
+
+        if (this.npmCheckIfPackageExists(package_name, package_version) == true) {
+            return false;
+        }
+
+        // If it is not found, then download it into our node_modules
+        try {
+            console.log("Could not find " + package_name);
+            console.log("Downloading " + package_name + " from npm for build...");
+            // Try to intall the given package and ONLY the given package. Will throw an exception if there is an error.
+            if (package_version != null) {
+                child_process.execSync("npm install --no-package-lock --ignore-scripts " + package_name + "@" + package_version);
+            } else {
+                child_process.execSync("npm install --no-package-lock --ignore-scripts " + package_name);
+            }
+            return true;
+
+        } catch (err) {
+            console.log("ERROR - npm could not download " + package_name + "! " + package_name + " is required to build the CRT");
+            throw err;
+        }
+    },
+
+    /**
+     * Tells NPM to uninstall a package. This should only be used to clean up a dynamic package downloaded with the
+     * npmDownloadAndInstallRuntimePackage function, as otherwise it could remove a non-dynamic package.
+     * @param {*} package_name The name of the package you want to delete (example 'cmake-js')
```

Comment:
minor issue: these `{*}` are supposed to be type annotations like `@param {string} package_name`